ci_scripts/meson-build.sh

```diff
@@ -17,5 +17,31 @@ source "$SCRIPT_DIR/env.sh"
 
 cd "$SCRIPT_DIR/.."
 
-meson setup build --prefix "$INSTALL_DIR" --buildtype="$1" -Dcassert=true -Dtap_tests=enabled $ENABLE_COVERAGE
+BUILD_TYPE=
+
+case "$1" in
+    debug)
+        echo "Building with debug option"
+        BUILD_TYPE=$1
+        ;;
+
+    debugoptimized)
+        echo "Building with debugoptimized option"
+        BUILD_TYPE=$1
+        ;;
+
+    sanitize)
+        echo "Building with sanitize option"
+        export CFLAGS="-fsanitize=address -fsanitize=undefined -fno-omit-frame-pointer -fno-inline-functions"
```

Comment:
Is there a reason why you disable inlining? LLVM's documentation says nothing about that as far as I can see.

https://clang.llvm.org/docs/UndefinedBehaviorSanitizer.html#stack-traces-and-report-symbolization

Reply:
https://clang.llvm.org/docs/AddressSanitizer.html#usage

> To get a reasonable performance add -O1 or higher. To get nicer stack traces in error messages add -fno-omit-frame-pointer. To get perfect stack traces you may need to disable inlining (just use -O1) and tail call elimination (-fno-optimize-sibling-calls).

Reply:
This is for ASAN. BTW it looks like it's worth to add `-fno-optimize-sibling-calls` as well.